### PR TITLE
Add pytest conftest.py file to suppress warnings with NumPy 2.0

### PR DESCRIPTION
**Pull request by seisman**
Workaround from https://github.com/GenericMappingTools/pygmt/issues/2628#issuecomment-1688666672.

Fixes https://github.com/GenericMappingTools/pygmt/issues/2628.

### CODE REVIEW

**Review by michaelgrund — APPROVED** [review]

File: pygmt/conftest.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+"""
+conftets.py for pytest.
```
Comment (weiji14) [inline pygmt/conftest.py:2]:
```suggestion
conftest.py for pytest.
```

File: pygmt/conftest.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+"""
+
+import numpy as np
+from packaging.version import Version
+
+# Keep this until we require numpy to be >=2.0
+# Address https://github.com/GenericMappingTools/pygmt/issues/2628.
```
Comment (weiji14) [inline pygmt/conftest.py:9]: What do we do though once we require numpy>2.0 in June 2026?

Comment (seisman) [reply pygmt/conftest.py:9]: At that time, this file can be removed and we just need to update the doctests to match the numpy 2.0 behavior.

File: pygmt/conftest.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+conftest.py for pytest.
+"""
+
+import numpy as np
+from packaging.version import Version
+
+# Keep this until we require numpy to be >=2.0
+# Address https://github.com/GenericMappingTools/pygmt/issues/2628.
+if Version(np.__version__) >= Version("2.0.0.dev0+git20230726"):
+    np.set_printoptions(legacy="1.25")
```
Comment (seisman) [inline pygmt/conftest.py:11]: https://github.com/numpy/numpy/blob/1d49c7f7ff527c696fc26ab2278ad51632a66660/numpy/_core/arrayprint.py#L91
`legacy` can be `1.25` but it's not well documented at https://numpy.org/doc/stable/reference/generated/numpy.set_printoptions.html#numpy.set_printoptions, so we need to ignore the type error:

```suggestion
    np.set_printoptions(legacy="1.25")  # type: ignore[arg-type]
```

**Pull request merged**